packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -306,19 +309,34 @@ export class TeamsBotV2Impl {
   private resolveScenarios(
     actRoles: PluginActRoles[],
     triggers: BotTrigger[],
-    hostType: HostType
+    hostType: HostType,
+    isM365: boolean | undefined
   ): string[] {
     const scenarios: string[] = [];
-    if (hostType === HostType.Function) {
-      if (actRoles.includes(PluginActRoles.Notification)) {
-        scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
-        triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+    actRoles.map((actRole) => {
+      switch (actRole) {
+        case PluginActRoles.CommandAndResponse:
+          scenarios.push(TemplateProjectsScenarios.COMMAND_AND_RESPONSE_SCENARIO_NAME);
+          break;
+        case PluginActRoles.Notification:
+          if (hostType === HostType.Function) {
+            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_FUNCTION_BASE_SCENARIO_NAME);
+            triggers.map((trigger) => scenarios.push(TriggerTemplateScenarioMappings[trigger]));
+          }
+          if (hostType === HostType.AppService) {
+            scenarios.push(TemplateProjectsScenarios.NOTIFICATION_RESTIFY_SCENARIO_NAME);
+          }
+          break;
+        case PluginActRoles.Bot || PluginActRoles.MessageExtension:
```

Comment:
Do we need to support notification bot + message extension in the future?

Reply:
I added the logic In line 333. When bot and message extension are both selected, the legacy bot will be pushed only once/

Reply:
Get it. I just misunderstood.